workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/RenderAsState.kt

```diff
@@ -0,0 +1,206 @@
+@file:OptIn(ExperimentalWorkflowApi::class)
+
+package com.squareup.workflow1.ui.compose
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.RememberObserver
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.saveable.LocalSaveableStateRegistry
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.SaverScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import com.squareup.workflow1.ExperimentalWorkflowApi
+import com.squareup.workflow1.Snapshot
+import com.squareup.workflow1.TreeSnapshot
+import com.squareup.workflow1.Workflow
+import com.squareup.workflow1.WorkflowInterceptor
+import com.squareup.workflow1.renderWorkflowIn
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import okio.ByteString
+
+/**
+ * Runs this [Workflow] as long as this composable is part of the composition, and returns a
+ * [State] object that will be updated whenever the runtime emits a new [RenderingT].
+ *
+ * The workflow runtime will be started when this function is first added to the composition, and
+ * cancelled when it is removed or if the composition fails. The first rendering will be available
+ * immediately as soon as this function returns as [State.value]. Composables that read this value
+ * will automatically recompose whenever the runtime emits a new rendering. In most cases you
+ * probably want to pass the rendering value to the [WorkflowRendering] composable.
```

Comment:
```suggestion
 * will automatically recompose whenever the runtime emits a new rendering. If you are driving UI form the Workflow tree invoked by [renderAsState] then you will
 * probably want to pass the State's value (the Workflow rendering) to the [WorkflowRendering] composable.
```

Reply:
There is just so much overloading going on between Compose/Workflow terms that I don't think you could be over explicit.